Jenkinsfile.promote

```diff
@@ -107,17 +107,30 @@ pipeline {
                 }
             }
         }
-        stage('Git tag') {
+        stage('Git tag and release CLI') {
             when {
                 expression {
-                    return getGitTag() != ""
+                    return isRelease() && getGitTag() != ""
                 }
             }
             steps {
                 script {
-                    sh "git pull origin ${getBuildBranch()}"
-                    githubscm.tagRepository(getGitTag(), env.BUILD_TAG)
-                    githubscm.pushObject("origin", "--tags ${getGitTag()}")
+                    sh "./hack/go-build-cli.sh true ${getGitTag()}"
```

Comment:
we should use `getProjectVersion` here I think

Reply:
I was unsure of whether to use the tag or project version. But, I changed it so that it uses the project version for the CLI file names and release title/description and the tag only for the `--tag` flag in `github-release`.